### ISSUE TITLE
Numeric type value is converting to string

### ISSUE THREAD

**Issue by majkelwork** [revolist/revogrid#485]
**Check if issue exists**
Not exists, in revogrid version 2 is working

**Describe the issue**
I am using [column numeral plugin](https://github.com/revolist/Revogrid-column-numeral) and my data contains numbers with column type 'numeric'. When I edit a row in the datagrid, the value type is changing from `number` to `string`

**To Reproduce**
[Codesandbox](https://codesandbox.io/p/sandbox/vue-grid-date-plugin-forked-xx6rg2?file=%2Fsrc%2FApp.vue%3A10%2C9])

**Comment by revolist** [comment]
Certainly! Here’s a concise and structured response you can use for a GitHub issue:

---

Hi @majkelwork ,

Thank you for reporting this issue. We've investigated the problem and found that it’s related to how the numeric column type is currently implemented.

### Issue Description

In the Revogrid version you're using, when editing a cell in a column with the type 'numeric', the value type changes from a number to a string. This is due to the editor component using `HTMLInputElement.value`, which captures the input as a string.

### Findings

- **Current Behavior:** The numeric column type only handles rendering, and the editor doesn’t automatically convert the string input to a number. This was intended to work this way from the beginning.
- **Version 2 Behavior:** It’s possible that version 2 handled this differently, possibly due to the input returning a number by default.

### Solution

To properly handle numeric values, we recommend manually parsing the input value to convert it to a number with `beforeedit` event. 

### Future Plans

We plan to improve this by automatically applying `parseFloat` for numeric inputs in future updates which would require use to create separate editor for number components. However, this feature is not currently prioritized, so for now, handling the conversion manually is the best approach.

### Summary

- **Issue:** Numeric column values are treated as strings after editing.
- **Current Handling:** Input values are captured as strings.
- **Workaround:** Manually parse the input using `parseFloat`.
- **Future Update:** We’ll consider adding automatic conversion to number in a future release.

We appreciate your understanding and patience. If you have any more questions or need further assistance, please let us know!

Best regards,  
Revolist Team

**Comment by majkelwork** [comment]
@revolist 
What is then the point of using columnTypes?
BeforeEdit is not a workaround, because the grid will still store data as string instead of number. Grid recognizes that an edit occurred when the change from number to string occurred. It causes excessive triggering of afterEdit, and I am not able to decide if the row was edited or only clicked.
The biggest bug occurs when the number is 0, because then it is removed and replaced with an empty string after click on cell editor.

**Comment by m2a2x** [comment]
Hello @majkelwork I just fixed 0 case scenario for you, it's much bigger issue
